src/core/service.c

```diff
@@ -2939,6 +3078,9 @@ static int service_stop(Unit *u) {
                 service_kill_control_process(s);
```

Comment:
please remove this. redundant given it falls through

Reply:
sorry I don't follow - the SERVICE_REFRESH_EXTENSIONS case can't call service_live_mount_finish() as that's another thing, so it needs a separate case?

Reply:
The order in which kill_control_process and live_mount_finish are called doesn't matter. Hence you can just allow the next case to kill it and only call live_mount_finish here